conda_env/exceptions.py

```diff
@@ -7,3 +7,38 @@ def __init__(self, filename, *args, **kwargs):
         msg = '{} file not found'.format(filename)
         self.filename = filename
         super(EnvironmentFileNotFound, self).__init__(msg, *args, **kwargs)
+
+
+class EnvironmentFileDoesNotExist(CondaEnvException):
+    def __init__(self, handle, *args, **kwargs):
+        self.handle = handle
+        msg = "{} does not have an environment definition".format(handle)
+        super(EnvironmentFileDoesNotExist, self).__init__(msg, *args, **kwargs)
+
+
+class EnvironmentFileNotDownloaded(CondaEnvException):
+    def __init__(self, username, packagename, *args, **kwargs):
+        msg = '{}/{} file not downloaded'.format(username, packagename)
+        self.username = username
+        self.packagename = packagename
+        super(EnvironmentFileNotDownloaded, self).__init__(msg, *args, **kwargs)
+
+
+class EnvironmentUsernameRequired(CondaEnvException):
+    def __init__(self, *args, **kwargs):
+        msg = 'Username is required'
+        super(EnvironmentUsernameRequired, self).__init__(msg, *args, **kwargs)
+
+
+class PackageNotFound(CondaEnvException):
+    def __init__(self, username, packagename, *args, **kwargs):
+        msg = '{}/{} not found'.format(username, packagename)
+        self.username = username
+        self.packagename = packagename
+        super(PackageNotFound, self).__init__(msg, *args, **kwargs)
+
+
+class LoaderNotFound(CondaEnvException):
```

Comment:
In order for this to be caught by [`conda.cli.common.arg_func`][1] you need to also extend `RuntimeError`.  I'm fine with the inheritance being `LoaderNotFound` extends `CondaEnvRuntimeError` extends `CondaEnvException` and `RuntimeError`.  That provides a nice clean way to capture all `CondaEnvRuntimeError` exceptions.

[1]: https://github.com/conda/conda/blob/master/conda/cli/main.py#L209-L210